File: reto_clean_code/python/domain/carrito.py
"""
Carrito de compras asociado a un cliente.  Permite agregar ítems y calcular
el total del pedido.
"""

from typing import List
from .item_carrito import ItemCarrito
from .producto import Producto


class Carrito:
    def __init__(self, id_cliente: str) -> None:
        self.id_cliente = id_cliente
        self.items: List[ItemCarrito] = []

    def agregar_item(self, producto: Producto, cantidad: int) -> None:
        self.items.append(ItemCarrito(producto, cantidad))

    def calcular_total(self) -> float:
        return sum(item.producto.precio * item.cantidad for item in self.items)

File: reto_clean_code/angular/domain/carrito.ts
// Carrito de compras asociado a un cliente.  Permite agregar ítems y calcular
// el total del pedido.

import { ItemCarrito } from './itemCarrito';
import { Producto } from './producto';

export class Carrito {
  public items: ItemCarrito[] = [];

  constructor(public idCliente: string) {}

  agregarItem(producto: Producto, cantidad: number): void {
    this.items.push(new ItemCarrito(producto, cantidad));
  }

  calcularTotal(): number {
    return this.items.reduce((acc, item) => acc + item.producto.precio * item.cantidad, 0);
  }
}